crates/neon-runtime/src/napi/string.rs

```diff
@@ -44,3 +44,13 @@ pub unsafe extern "C" fn data(env: Env, out: *mut u8, len: isize, value: Local)
 
     read.assume_init() as isize
 }
+
+pub unsafe extern "C" fn run_script(out: &mut Local, env: Env, value: Local) -> bool {
+    let status = napi::run_script(
+        env,
+        value,
+        out as *mut _,
+    );
+
+    status == napi::Status::Ok
```

Comment:
What are the failure conditions on this? Do we need to be concerned with other errors besides `Ok` and `Throw`?

Reply:
According to [the implementation of `napi_run_script `](https://github.com/nodejs/node/blob/03806a0bb2f3336bcbf3a668e156993cc2582c71/src/js_native_api_v8.cc#L3036) all code paths that don't return `Ok` have called `napi_set_last_error`. The docs don't state clearly though.
For a peace of mind I've added a test for invalid syntax.

Reply:
Thanks for checking!